jobs/generation/Utilities.groovy

```diff
@@ -302,7 +306,50 @@ class Utilities {
     //  isPR: True if job is PR job, false otherwise.
     //  defaultBranch: If not a PR job, the branch that we should be building.
     def static standardJobSetup(def job, String project, boolean isPR, String defaultBranch = '*/master') {
-        Utilities.addStandardParameters(job, project, isPR, defaultBranch)
+        String defaultRefSpec = getDefaultRefSpec(null)
+        if (isPR) {
+            defaultBranch = getDefaultBranchOrCommitPR(null)
+        }
+        standardJobSetupEx(job, project, isPR, defaultBranch, defaultRefSpec)
+    }
+
+    // Performs standard job setup for a newly created push job.
+    // Includes: Basic parameters, Git SCM, and standard options
+    // 
+    // Parameters:
+    //  job: Job to set up.
+    //  project: Name of project
+    //  defaultBranch: Branch to build on push. 
+    def static standardJobSetupPush(def job, String project, String defaultBranch = null) {
+        defaultBranch = getDefaultBranchOrCommitPush(defaultBranch);
+        standardJobSetupEx(job, project, false, defaultBranch, null);
+    }
+
+    // Performs standard job setup for a newly created push job.
+    // Includes: Basic parameters, Git SCM, and standard options
+    // 
+    // Parameters:
+    //  job: Job to set up.
+    //  project: Name of project
+    //  defaultBranchOrCommit: Commit / branch to build. 
+    //  defaultRefSpec: the refs that Jenkins must sync on a PR job
+    def static standardJobSetupPR(def job, String project, String defaultBranchOrCommit = null, String defaultRefSpec = null) {
```

Comment:
Fine with removing defaults but how do you want consumers to opt into the default values?  Do you want them to manually call `getDefaultBranchOrCommit`? 

As for private I was actually thinking about making the Ex version private.  I found the `Push` vs. `PR` more declarative.  Especially lacking default parameters it will be less confusing to call 

```
standardJobSetupPush(...)
vs.
standardJobSetupEx(..., false, defaultBranch, whyDoesMyPushJobNeedRefSpec) 
```